contracts/src/Gateway.sol

```diff
@@ -76,19 +78,26 @@ contract Gateway is IGateway, IInitializable {
         address agentExecutor,
         uint256 dispatchGas,
         ParaID bridgeHubParaID,
-        bytes32 bridgeHubHubAgentID,
+        bytes32 bridgeHubAgentID,
         ParaID assetHubParaID,
-        bytes32 assetHubHubAgentID,
+        bytes32 assetHubAgentID,
         bytes2 createTokenCallID
     ) {
+        if (bridgeHubParaID == assetHubParaID) {
+            revert BridgeHubAssetHubParaIDsIdentical();
```

Comment:
I wouldn't really bother with having very specific error types. Just raising `InvalidConstructParams` or something like that should be sufficient

Reply:
Agreed, especially because this is part of the contract setup and not end-user functionality.